clippy_lints/src/time_subtraction.rs

```diff
@@ -132,23 +135,30 @@ fn print_manual_instant_elapsed_sugg(cx: &LateContext<'_>, expr: &Expr<'_>, sugg
     );
 }
 
-fn print_unchecked_duration_subtraction_sugg(
+fn print_unchecked_time_subtraction_sugg(
     cx: &LateContext<'_>,
     left_expr: &Expr<'_>,
     right_expr: &Expr<'_>,
     expr: &Expr<'_>,
 ) {
     let mut applicability = Applicability::MachineApplicable;
+    let ty = cx.typeck_results().expr_ty(expr).peel_refs();
+    let (left_default, left_ty) = match ty::get_type_diagnostic_name(cx, ty) {
+        Some(v) => (format!("<{}>", v.as_str().to_lowercase()), v.as_str().to_string()),
+        None => {
+            return;
+        },
+    };
```

Comment:
You can keep the article and the formatting with this change (and the next one):

```suggestion
    let (left_default, left_ty) = match ty::get_type_diagnostic_name(cx, ty) {
        Some(sym::Instant) => ("<instant>", "an `Instant`"),
        Some(sym::Duration) => ("<duration>", "a `Duration`"),
        _ => return,
    };
```